drivers/sensor/adi/ade9153a/ade9153a.c

```diff
@@ -0,0 +1,556 @@
+{
+	LOG_DBG("Autocalibrating Current Channel");
+	start_acal(dev, 0x00000017); /* AITurbo */
+	k_msleep(CONFIG_ADE9153A_AI_TURBO_CAL_TIME);
+	stop_acal(dev);
+
+	LOG_DBG("Autocalibrating Voltage Channel");
+	start_acal(dev, 0x00000043); /* AVTurbo */
+	k_msleep(CONFIG_ADE9153A_AV_TURBO_CAL_TIME);
+	stop_acal(dev);
+
+	double aicc, avcc;
+	uint32_t temp_reg;
+
+	ade9153a_get_reg32(dev, ADE9153A_REG_MS_ACAL_AICC, &temp_reg); /* turbo */
+
+	aicc = temp_reg / (double)2048;
+
+	ade9153a_get_reg32(dev, ADE9153A_REG_MS_ACAL_AVCC, &temp_reg); /* turbo */
+
+	avcc = temp_reg / (double)2048;
+
+	apply_acal(dev, aicc, avcc);
+
+	LOG_DBG("Autocalibration...[ok]");
+	k_msleep(100);
+
+	return 0;
+}
+
+static int ade9153a_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct ade9153a_data *data = dev->data;
+	double double_value;
+
+	switch (chan) {
+	case SENSOR_CHAN_DIE_TEMP: {
+		uint16_t gain;
+		uint16_t offset;
+
+		gain = (data->temperature_trim & 0xFFFF);           /* Extract 16 LSB */
+		offset = ((data->temperature_trim >> 16) & 0xFFFF); /* Extract 16 MSB */
+		double_value = ((double)offset / 32.00) -
+			       ((double)data->temperature_reg * (double)gain / (double)131072);
+		break;
+	}
+	case SENSOR_CHAN_AC_ACTIVE_ENERGY: {
+		/* Energy in mWhr */
+		double_value = (double)data->active_energy_reg * CAL_ENERGY_CC / 1000;
+
+		break;
+	}
+	case SENSOR_CHAN_AC_FUNDAMENTAL_REACTIVE_ENERGY: {
+		double_value = (double)data->fund_reactive_energy_reg * CAL_ENERGY_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_APPARENT_ENERGY: {
+		double_value = (double)data->apparent_energy_reg * CAL_ENERGY_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_ACTIVE_POWER: {
+		double_value = (double)data->active_power_reg * CAL_POWER_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_FUNDAMENTAL_REACTIVE_POWER: {
+		double_value = (double)data->fund_reactive_power_reg * CAL_POWER_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_APPARENT_POWER: {
+		double_value = (double)data->apparent_power_reg * CAL_POWER_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_CURRENT_RMS: {
+		/* RMS in mA */
+		double_value = (double)data->current_rms_reg * CAL_IRMS_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_HALF_CURRENT_RMS: {
+		/* RMS in mV */
+		double_value = (double)data->half_current_rms_reg * CAL_IRMS_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_VOLTAGE_RMS: {
+		/* Half-RMS in mV */
+		double_value = (double)data->voltage_rms_reg * CAL_VRMS_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_HALF_VOLTAGE_RMS: {
+		/* Half-RMS in mV */
+		double_value = (double)data->half_voltage_rms_reg * CAL_VRMS_CC / 1000;
+		break;
+	}
+	case SENSOR_CHAN_AC_POWER_FACTOR: {
+		/* Calculate PF */
+		double_value = (double)data->power_factor_reg / (double)134217728;
+		break;
+	}
+	case SENSOR_CHAN_AC_PERIOD: {
+		/* Calculate Frequency */
+		double_value = (double)(4000 * 65536) / (double)(data->period_reg + 1);
```

Comment:
@rodrigopex In the [Datasheet Page 27](https://www.analog.com/media/en/technical-documentation/data-sheets/ade9153a.pdf), it gives the `Line Period Calculation` as the inverse of what you have here. 

It looks like both your `SENSOR_CHAN_AC_PERIOD`  & `SENSOR_CHAN_AC_FREQUENCY` give the same value with Frequency just scaled by `100`.

Should it not look like:
```
case SENSOR_CHAN_AC_PERIOD: {
	/* Calculate Line Period */
	double_value = (double)(data->period_reg + 1) / (double)(4000 * 65536);
	break;
}
case SENSOR_CHAN_AC_FREQUENCY: {
	/* Calculate Frequency in cHz*/
	double_value = (4000LU * 65536 * 100) / (double)(data->period_reg + 1);
	break;
}
```

Reply:
Ok. I will fix that.

Reply:
Done.